source/tutorials/installing-nixos-on-a-raspberry-pi.rst

```diff
@@ -0,0 +1,184 @@
+
+
+Updating firmware
+-----------------
+
+To benefit from updates and bug fixes from the vendor, we'll start by updating Raspberry Pi firmware:
+
+.. code:: shell-session
+
+  # nix-shell -p raspberrypi-eeprom
+  # FIRMWARE_RELEASE_STATUS=stable rpi-eeprom-update -d -a
+
+  
+Installing NixOS 
+----------------
+
+For initial installation we'll install `XFCE <https://www.xfce.org/>`_ desktop environment
+with user ``guest`` and SSH daemon.
+
+.. code:: nix 
+
+  { config, pkgs, lib, ... }:
+
+  let
+    user = "guest";
+    password = "guest";
+    SSID = "mywifi";
+    SSIDpassword = "mypassword";
+    interface = "wlan0";
+    hostname = "myhostname";
+  in {
+    imports = ["${fetchTarball "https://github.com/domenkozar/nixos-hardware/archive/rpi4.tar.gz" }/raspberry-pi/4"];
+
+    fileSystems = {
+      "/" = {
+        device = "/dev/disk/by-label/NIXOS_SD";
+        fsType = "ext4";
+        options = [ "noatime" ];
+      };
+    };
+
+    networking = {
+      hostName = hostname;
+      wireless = {
+        enable = true;
+        networks."${SSID}".psk = SSIDpassword;
+        interfaces = [ interface ];
+      };
+    };
+
+    environment.systemPackages = with pkgs; [ vim ];
+
+    services.openssh.enable = true;
+
+    users = {
+      mutableUsers = false;
+      users.guest = {
+        isNormalUser = true;
+        password = password;
+        extraGroups = [ "wheel" ];
+      };
+    };
+
+    # video
+    services.xserver = {
+      enable = true;
+      displayManager.lightdm.enable = true;
+      desktopManager.xfce.enable = true;
+      videoDrivers = [ "fbdev" ];
+    };
+
+    # audio
+    hardware.pulseaudio.enable = true;
+  }
+
+To save time on typing the whole configuration, download it:
+
+.. code:: shell-session
+
+  # curl -L https://tinyurl.com/nixos-rpi-tutorial-preview > /etc/nixos/configuration.nix 
+
+At the top of `/etc/nixos/configuration.nix` there are a few variables that you want to configure,
+most important being your wifi connection details, this time specified in declarative way.
+
+Once you're ready to install NixOS:
+
+.. code:: shell-session
+
+  # nixos-install --root /
+  # reboot
+
+In case your system doesn't boot, select the oldest configuration in the bootloader menu to get back to live image and start over.
+
+
+Making changes 
+--------------
+
+It booted, congratulations!
+
+To make further changes to the configuration, `search through NixOS options <https://search.nixos.org/options>`_,
+edit ``/etc/nixos/configuration.nix`` and update your system:
+
+.. code:: shell-session 
+
+  $ sudo -i
+  # nixos-rebuild switch
+
+
+Going forward
+-------------
+
+- Once you have successfully running OS, try upgrading it with `nixos-rebuild switch --upgrade` and boot the old configuration if something broke.
+  
+- To tweak bootloader options affecting hardware, look at `setting /boot/config.txt options <https://www.raspberrypi.org/documentation/configuration/config-txt/>`_.
```

Comment:
I'm not sure how it would be integrated to your tutorial, but it might be important to specify that `config.txt` will not be found under `/boot/` like it would in most Raspberry Pi-first distros.

That to edit the `config.txt` options, `/dev/disk/by-label/FIRMWARE` needs to be (temporarily) mounted to `/mnt` (or elsewhere).